zipline/pipeline/loaders/quarter_estimates.py

```diff
@@ -0,0 +1,391 @@
+        """
+        requested_qtr_data = stacked_last_per_qtr.loc[idx]
+        # We no longer need this in the index, but we do need it as a column
+        # to calculate adjustments.
+        requested_qtr_data = requested_qtr_data.reset_index(
+            SHIFTED_NORMALIZED_QTRS
+        )
+        (requested_qtr_data[FISCAL_YEAR_FIELD_NAME],
+         requested_qtr_data[FISCAL_QUARTER_FIELD_NAME]) = \
+            split_normalized_quarters(
+                requested_qtr_data[SHIFTED_NORMALIZED_QTRS]
+            )
+        # Move sids into the columns. Once we're left with just dates
+        # as the index, we can reindex by all dates so that we have a
+        # value for each calendar date.
+        requested_qtr_data = requested_qtr_data.unstack(
+            SID_FIELD_NAME
+        ).reindex(dates)
+        return requested_qtr_data
+
+    def get_adjustments(self,
+                        zero_qtr_idx,
+                        requested_qtr_idx,
+                        stacked_last_per_qtr,
+                        last_per_qtr,
+                        dates,
+                        column_name,
+                        column,
+                        mask,
+                        assets,
+                        qtr_crossover_point):
+        """
+        Creates an AdjustedArray from the given estimates data for the given
+        dates.
+
+        Parameters
+        ----------
+        zero_qtr_idx : pd.MultiIndex
+            The index of the row of the zeroth (immediately next/previous)
+            quarter from each date for each sid.
+        requested_qtr_idx : pd.MultiIndex
+            The index of the row of the requested quarter from each date for
+            each sid.
+        stacked_last_per_qtr : pd.DataFrame
+            The latest estimate known per sid per date per quarter with the
+            dates, normalized quarter, and sid as the index.
+        last_per_qtr : pd.DataFrame
+            The latest estimate known per sid per date per quarter with
+            dates as the index and normalized quarter and sid in the columns
+            MultiIndex; allows easy access to the timeline of estimates
+            across all dates for a sid for a particular quarter.
+        dates : pd.DatetimeIndex
+            The calendar dates for which estimates data is requested.
+        column_name : string
+            The name of the column for which the AdjustedArray is being
+            computed.
+        column : BoundColumn
+            The column for which the AdjustedArray is being computed.
+        mask :
+        assets :
+        qtr_crossover_point :
+            Whether we should use the 'right' or 'left' side when doing
+            searchsorted on the dates for quarter boundaries.
+
+        Returns
+        -------
+        adjusted_array : AdjustedArray
+            The array of data and overwrites for the given column.
+        """
+        adjustments = defaultdict(list)
```

Comment:
I guess it doesn't have to be, since we're not extending the list for any key at any point.